src/graph_synchronizer.cpp

```diff
@@ -122,37 +119,36 @@ void GraphSynchronizer::Lock::lock() {
             // Make them into pos_ts that point left to right, the way Jordan thinks.
             pos_t left_pos = make_pos_t(start_left.node, start_left.is_end, 0);
             pos_t right_pos = make_pos_t(end_right.node, !end_right.is_end,
-                synchronizer.graph.get_node(end_right.node)->sequence().size() - 1);
+                synchronizer.graph.get_node(end_right.node)->sequence().size());
             
             // Since these are already at node ends, we don't need to worry about node cuts.
             
             // Extract paths out to the length we need to connect the ends, or a bit further.
             // TODO: be sure to extract really big indels somehow...
-            auto duplications = algorithms::extract_connecting_graph(&synchronizer.graph,
-                context_graph,
+            auto translator = algorithms::extract_connecting_graph(&synchronizer.graph,
+                &context,
                 (past_end - start) * 2,
                 left_pos,
                 right_pos,
-                true, // Keep the specified positions in the graph
                 false, // Disallow terminal node cycles, so we don't duplicate nodes
-                true, // Remove other tips
-                false, // We don't care about being on a path
-                false); // Or being strictly less than the specified length
+                true, // We don't want extraneous material that doesn't connect the positions
+                false); // But we don't care about being strictly less than the specified length
                 
 #ifdef debug
             cerr << "Extracted " << context_graph.node_size() << " nodes and " << context_graph.edge_size() << " edges between " << path_name << ":" << start << "-" << past_end << endl;
```

Comment:
This line ought to change from context_graph to context, or things will go wrong if debug is defined.